.github/workflows/go.yml

```diff
@@ -1,23 +1,43 @@
-name: i386 linux tests
+name: Run unit tests on Linux
 
 on:
   push:
-    branches: [ master ]
+    branches: [ main ]
   pull_request:
-    branches: [ master ]
+    branches: [ main ]
   workflow_dispatch:
 
+env:
+  CGO_CFLAGS_ALLOW: "-O -D__BLST_PORTABLE__"
+  CGO_CFLAGS: "-O -D__BLST_PORTABLE__"
+
 jobs:
   build:
-    runs-on: self-hosted
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.20', '1.21.x' ]
+        goarch: ["386", amd64, arm64]
```

Comment:
note- this is going to do 6 runs each for each PR, so let's keep an eye on performance.